packages/jest-worker/src/workers/__tests__/WorkerEdgeCases.test.js

```diff
@@ -106,174 +101,298 @@ describe.each([
   }
 
   test('should get memory usage', async () => {
-    worker = new workerClass({
-      childWorkerPath: workerPath,
-      maxRetries: 0,
-      workerPath: join(__dirname, '__fixtures__', 'EdgeCasesWorker'),
-    });
+    let worker;
+
+    try {
+      worker = new workerClass({
+        childWorkerPath: workerPath,
+        maxRetries: 0,
+        workerPath: join(__dirname, '__fixtures__', 'EdgeCasesWorker'),
+      });
 
-    const memoryUsagePromise = worker.getMemoryUsage();
-    expect(memoryUsagePromise).toBeInstanceOf(Promise);
+      const memoryUsagePromise = worker.getMemoryUsage();
+      expect(memoryUsagePromise).toBeInstanceOf(Promise);
 
-    expect(await memoryUsagePromise).toBeGreaterThan(0);
+      expect(await memoryUsagePromise).toBeGreaterThan(0);
+    } finally {
+      if (worker) {
+        worker.forceExit();
+        await worker.waitForExit();
+      }
+    }
```

Comment:
instead of wrapping all tests in `try-finally`, should we create a helper function which is just

```ts
async function closeWorkerAfter(worker: Worker, testBody: () => Promise<void>) {
  try {
    await testBody();
  } finally {
    worker.forceExit();
    await worker.waitForExit();
  }
}
```

called like

```ts
test('should get memory usage', async () => {
  const worker = new workerClass({
    childWorkerPath: workerPath,
    maxRetries: 0,
    workerPath: join(__dirname, '__fixtures__', 'EdgeCasesWorker'),
  });
  await closeWorkerAfter(worker, async () => {
    const memoryUsagePromise = worker.getMemoryUsage();
    expect(memoryUsagePromise).toBeInstanceOf(Promise);

    expect(await memoryUsagePromise).toBeGreaterThan(0);
  });
});
```

Reply:
Nice!